static/js/message_events.js

```diff
@@ -221,9 +234,6 @@ exports.update_messages = function update_messages(events) {
 
         if (event.orig_content !== undefined) {
             if (page_params.realm_allow_edit_history) {
-                // Most correctly, we should do this for topic edits as
-                // well; but we don't use the data except for content
-                // edits anyway.
```

Comment:
So, I think if we want to match the behavior the server would give if you received a message where edits had happened in the past, if an event has both a topic and a message edit, we should be appending a single `edit_history_event`, not two.  

So I think the correct behavior is to extract this block modifying `msg.edit_history` to a common that sits above the `topic_edit` conditional (and just copies keys from the event; there's probably a good `_` function that would do this really cleanly)